enum.c

```diff
@@ -0,0 +1,71 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+char *s_gets(char * st, int n);
+enum spectrum {red, orange, yellow, green, blue, violet};
+const char * colors[] ={"red","orange","yellow","green","blue","violet"};
+#define LEN 30
+
+int main(void)
+{
+    char choice[LEN];
+    enum spectrum color;
+    bool color_is_found = false;
+
+    puts("Enter a color (empty line to quit):");
+    while(s_gets(choice, LEN) != NULL && choice[0] != '\0')
```

Comment:
```suggestion
    while((s_gets(choice, LEN) != NULL) && (choice[0] != '\0'))
```
More parens for clarity